.github/workflows/build.yml

```diff
@@ -50,7 +50,34 @@ jobs:
 
       - name: Expose JSON data
         run: cp build/data.json dist/data.json
+        
+      - name: Download old artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: build.yml
+          workflow_conclusion: completed
+          name: data.json
+          path: build/old.json
+
+      - name: Compare both artifacts
+        run: (diff build/data.json build/old.json &&  echo "NOT OK" | grep "NOT OK") && false # will exit the workflow with an error, stopping next build steps
```

Comment:
This will exit code 1 if the files match, which will cause the run to be flagged as failing. We need to cleanly stop the workflow if the files are the same (maybe using the `if` functionality of actions?).